pkg/controllers/dynakube/extension/secret.go

```diff
@@ -98,3 +75,82 @@ func (r *reconciler) buildSecret(eecToken dttoken.Token, otelcToken dttoken.Toke
 func (r *reconciler) getSecretName() string {
 	return r.dk.Extensions().GetTokenSecretName()
 }
+
+func (r *reconciler) newSecret() (*corev1.Secret, error) {
+	newEecToken, err := dttoken.New(eecConsts.TokenSecretValuePrefix)
+	if err != nil {
+		log.Info("failed to generate eec token")
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, errors.Wrap(err, "error generating eec token"))
+
+		return nil, err
+	}
+
+	newOtelcToken, err := dttoken.New(consts.DatasourceTokenSecretValuePrefix)
+	if err != nil {
+		log.Info("failed to generate otelc token")
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, errors.Wrap(err, "error generating otelc token"))
+
+		return nil, err
+	}
+
+	newSecret, err := r.buildSecret([]byte(newEecToken.String()), []byte(newOtelcToken.String()))
+	if err != nil {
+		log.Info("failed to generate extension secret")
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, err)
+
+		return nil, err
+	}
+
+	return newSecret, nil
+}
+
+func (r *reconciler) migrateSecret(existingSecret *corev1.Secret) (*corev1.Secret, error) {
+	const oTelcTokenKey = "otelc.token"
+
+	if _, exists := existingSecret.Data[consts.DatasourceTokenSecretKey]; exists {
+		// no migration needed, datasource token file name already correct
+		return nil, ErrNoMigration
+	}
+
+	datasourceToken, exists := existingSecret.Data[oTelcTokenKey]
+	if !exists {
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, errors.Errorf("error migrating datasource token, missing %s key", oTelcTokenKey))
+
+		return nil, errors.New("missing otelc token in existing secret, cannot migrate")
+	}
+
+	eecToken, exists := existingSecret.Data[eecConsts.TokenSecretKey]
+	if !exists {
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, errors.Errorf("error migrating EEC token, missing %s key", eecConsts.TokenSecretKey))
+
+		return nil, errors.New("missing EEC token in existing secret, cannot migrate")
+	}
+
+	migratedSecret, err := r.buildSecret(eecToken, datasourceToken)
+	if err != nil {
+		conditions.SetSecretGenFailed(r.dk.Conditions(), secretConditionType, err)
+
+		return nil, err
+	}
+
+	return migratedSecret, nil
+}
+
+func (r *reconciler) cleanupSecret(ctx context.Context) {
```

Comment:
this function name is not accurate, it does more then `cleanupSecret` it does:
- `remove status condition` and 
- `remove secret`